--- a/src/util-prop-parsers.c
+++ b/src/util-prop-parsers.c
@@ -526,7 +526,7 @@
        while (current && *current != '\0' && bit <= ABS_MASK_FLAT) {
                if (*current != ':') {
                        int v;
-                       next = index(current, ':');
+                       next = strchr(current, ':');
                        if (next)
                                *next = '\0';


--- a/src/util-strings.c
+++ b/src/util-strings.c
@@ -383,7 +383,7 @@
        if (base == NULL)
                return safe_strdup("");

-       suffix = rindex(base, '.');
+       suffix = strrchr(base, '.');
        if (suffix == NULL)
                return safe_strdup(base);
        else
