--- a/src/kcrash.cpp
+++ b/src/kcrash.cpp
@@ -462,7 +462,7 @@ void KCrash::defaultCrashHandler(int sig)
         printf("BEES17\n");
         if ((s_flags & AutoRestart) && s_autoRestartCommandLine) {
             printf("BEES18\n");
-            QThread::sleep(1);
+            QThread::sleep(5);
             startProcess(s_autoRestartCommandLine.argc, const_cast<const char **>(s_autoRestartCommandLine.argv), false);
         }
         printf("BEES19\n");
