--- a/src/kcrash.cpp
+++ b/src/kcrash.cpp
@@ -466,7 +466,9 @@ void KCrash::defaultCrashHandler(int sig)
             startProcess(s_autoRestartCommandLine.argc, const_cast<const char **>(s_autoRestartCommandLine.argv), false);
         }
         printf("BEES19\n");
-        crashRecursionCounter++;
+        if (!crashRecurtionCounter) {
+            crashRecursionCounter++;
+        }
     }
 
     printf("BEES20\n");
