--- a/src/workspace.cpp  2025-11-12 19:54:40.924015222 +0530
+++ b/src/workspace.cpp  2025-11-13 06:16:35.785651603 +0530
@@ -1966,8 +1966,8 @@

 Window *Workspace::findWindow(const QUuid &internalId) const
 {
-    return findWindow([internalId](const KWin::Window *l) -> bool {
-        return internalId == l->internalId();
+    return findWindow([internalId](const Window *l) -> bool {
+        return l->internalId() == internalId;
     });
 }

--- a/src/workspace.h    2025-11-12 19:54:40.948015222 +0530
+++ b/src/workspace.h    2025-11-15 10:41:03.317924863 +0530
@@ -807,19 +807,6 @@
     return should_get_focus.count() > 0 ? should_get_focus.last() : m_activeWindow;
 }

-#if KWIN_BUILD_X11
-inline void Workspace::addGroup(Group *group)
-{
-    Q_EMIT groupAdded(group);
-    groups.append(group);
-}
-
-inline void Workspace::removeGroup(Group *group)
-{
-    groups.removeAll(group);
-}
-#endif
-
 inline const QList<Window *> &Workspace::stackingOrder() const
 {
     // TODO: Q_ASSERT( block_stacking_updates == 0 );
@@ -867,5 +854,18 @@
     return Workspace::_self;
 }

-} // namespace
+} // namespace KWin
 Q_DECLARE_OPERATORS_FOR_FLAGS(KWin::Workspace::ActivityFlags)
+
+#if KWIN_BUILD_X11
+inline void KWin::Workspace::addGroup(Group *group)
+{
+    Q_EMIT groupAdded(group);
+    groups.append(group);
+}
+
+inline void KWin::Workspace::removeGroup(Group *group)
+{
+    groups.removeAll(group);
+}
+#endif
