diff --git a/CMakeLists.txt b/CMakeLists.txt
index 59ddc5a..c7a2001 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,7 +75,7 @@ find_package(PlasmaActivitiesStats ${PROJECT_DEP_VERSION} REQUIRED)
 find_package(KSysGuard ${PROJECT_DEP_VERSION} CONFIG)
 set_package_properties(KSysGuard PROPERTIES
     DESCRIPTION "Components to monitor the system"
-    TYPE REQUIRED
+    TYPE OPTIONAL
 )
 find_package(KF6Screen ${PROJECT_DEP_VERSION} CONFIG REQUIRED)
 find_package(LayerShellQt ${PROJECT_DEP_VERSION} CONFIG REQUIRED)
@@ -125,7 +125,6 @@ set_package_properties(QALCULATE PROPERTIES
 pkg_check_modules(DBus dbus-1 REQUIRED IMPORTED_TARGET)
 
 find_package(KWinDBusInterface CONFIG REQUIRED)
-find_package(ScreenSaverDBusInterface CONFIG REQUIRED)
 
 # Used in kcm_autostart
 pkg_check_modules(SYSTEMD "systemd")
@@ -200,8 +199,6 @@ if(WITH_X11)
 
     find_package(XCB MODULE REQUIRED COMPONENTS AUX XCB CURSOR RANDR IMAGE)
     set_package_properties(XCB PROPERTIES TYPE REQUIRED)
-
-    find_package(KScreenLocker ${PROJECT_DEP_VERSION} REQUIRED)
 else()
     set(HAVE_X11 0)
     set(HAVE_XCURSOR 0)
@@ -320,7 +317,7 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     add_feature_info(GLIBC_LOCALE_GEN GLIBC_LOCALE_GEN "Generate glibc locales")
     add_feature_info(GLIBC_LOCALE_PREGENERATED GLIBC_LOCALE_PREGENERATED "glibc locales are pregenerated")
     ################## Find libraries ###################
-    if(REGION_LANG_GENERATE_LOCALE_HELPER)
+    if(FALSE)
         find_package(PolkitQt6-1)
         set_package_properties(PolkitQt6-1
                 PROPERTIES DESCRIPTION "DBus interface wrapper for Polkit"
diff --git a/components/sessionsprivate/CMakeLists.txt b/components/sessionsprivate/CMakeLists.txt
index dec965b..174def7 100644
--- a/components/sessionsprivate/CMakeLists.txt
+++ b/components/sessionsprivate/CMakeLists.txt
@@ -3,7 +3,6 @@ set(sessionsprivateplugin_SRCS
     types.h
 )
 
-qt_add_dbus_interface(sessionsprivateplugin_SRCS ${SCREENSAVER_DBUS_INTERFACE} screensaver_interface)
 
 kconfig_add_kcfg_files(sessionsprivateplugin_SRCS kscreensaversettings.kcfgc)
 
diff --git a/containmentactions/contextmenu/CMakeLists.txt b/containmentactions/contextmenu/CMakeLists.txt
index 751032f..f7aa88b 100644
--- a/containmentactions/contextmenu/CMakeLists.txt
+++ b/containmentactions/contextmenu/CMakeLists.txt
@@ -10,7 +10,6 @@ set(contextmenu_SRCS
 set(krunner_xml ${plasma-workspace_SOURCE_DIR}/krunner/dbus/org.kde.krunner.App.xml)
 qt_add_dbus_interface(contextmenu_SRCS ${krunner_xml} krunner_interface)
 
-qt_add_dbus_interface(contextmenu_SRCS ${SCREENSAVER_DBUS_INTERFACE} screensaver_interface)
 
 kcoreaddons_add_plugin(plasma_containmentactions_contextmenu SOURCES ${contextmenu_SRCS} INSTALL_NAMESPACE "plasma/containmentactions")
 
diff --git a/kcms/region_language/localegenhelper/CMakeLists.txt b/kcms/region_language/localegenhelper/CMakeLists.txt
index 21389d1..5858de8 100644
--- a/kcms/region_language/localegenhelper/CMakeLists.txt
+++ b/kcms/region_language/localegenhelper/CMakeLists.txt
@@ -17,7 +17,6 @@ qt_add_dbus_adaptor(localegen_Helper_SRCS ${CMAKE_CURRENT_BINARY_DIR}/org.kde.lo
 add_executable(plasma-localegen-helper ${localegen_Helper_SRCS})
 target_link_libraries(plasma-localegen-helper
     Qt::DBus
-    PolkitQt6-1::Core
     KF6::I18n
 )
 ecm_install_configured_files(INPUT org.kde.localegenhelper.service.in DESTINATION ${KDE_INSTALL_DBUSSYSTEMSERVICEDIR} @ONLY)
diff --git a/ksmserver/CMakeLists.txt b/ksmserver/CMakeLists.txt
index 1a9a7b5..ecb70be 100644
--- a/ksmserver/CMakeLists.txt
+++ b/ksmserver/CMakeLists.txt
@@ -36,7 +36,6 @@ configure_package_config_file(KSMServerDBusInterfaceConfig.cmake.in
     INSTALL_DESTINATION ${CMAKECONFIG_INSTALL_DIR})
 
 target_link_libraries(ksmserver
-    PW::KScreenLocker
     PW::KWorkspace
     KF6::XmlGui
     KF6::I18n
diff --git a/libkworkspace/CMakeLists.txt b/libkworkspace/CMakeLists.txt
index 02d8903..d78e221 100644
--- a/libkworkspace/CMakeLists.txt
+++ b/libkworkspace/CMakeLists.txt
@@ -21,7 +21,6 @@ set(kworkspace_LIB_SRCS kdisplaymanager.cpp
 
 add_definitions(-DTRANSLATION_DOMAIN=\"libkworkspace\")
 
-qt_add_dbus_interface(kworkspace_LIB_SRCS ${SCREENSAVER_DBUS_INTERFACE} screenlocker_interface )
 qt_add_dbus_interface(kworkspace_LIB_SRCS ${plasma-workspace_SOURCE_DIR}/ksmserver/org.kde.LogoutPrompt.xml logoutprompt_interface)
 qt_add_dbus_interface(kworkspace_LIB_SRCS ${plasma-workspace_SOURCE_DIR}/startkde/plasma-session/org.kde.Startup.xml startup_interface)
 qt_add_dbus_interface(kworkspace_LIB_SRCS ${plasma-workspace_SOURCE_DIR}/startkde/plasma-shutdown/org.kde.Shutdown.xml shutdown_interface)
diff --git a/runners/sessions/CMakeLists.txt b/runners/sessions/CMakeLists.txt
index 23ee4e2..068b8b2 100644
--- a/runners/sessions/CMakeLists.txt
+++ b/runners/sessions/CMakeLists.txt
@@ -6,7 +6,6 @@ set(krunner_sessions_SRCS
     sessionrunner.cpp sessionrunner.h
 )
 
-qt_add_dbus_interface(krunner_sessions_SRCS ${SCREENSAVER_DBUS_INTERFACE} screensaver_interface)
 
 kcoreaddons_add_plugin(krunner_sessions SOURCES ${krunner_sessions_SRCS} INSTALL_NAMESPACE "kf6/krunner")
 target_link_libraries(krunner_sessions Qt::DBus KF6::Runner KF6::I18n PW::KWorkspace)
