--- a/lib/plugins/SignOn/authpluginif.h
+++ b/lib/plugins/SignOn/authpluginif.h
@@ -37,9 +37,10 @@
 #include <SignOn/uisessiondata.h>
 #include <SignOn/signonerror.h>

+#include <QStringList>
+
 QT_BEGIN_NAMESPACE
 class QString;
-class QStringList;
 class QByteArray;
 class QVariant;
 QT_END_NAMESPACE
@@ -73,7 +74,7 @@
         }

 #define SIGNON_DECL_AUTH_PLUGIN(pluginclass) \
-        Q_EXTERN_C AuthPluginInterface *auth_plugin_instance() \
+        extern "C" AuthPluginInterface *auth_plugin_instance() \
         SIGNON_PLUGIN_INSTANCE(pluginclass)

 /*!
