--- a/lib/plugins/SignOn/authpluginif.h
+++ b/lib/plugins/SignOn/authpluginif.h
@@ -28,6 +28,7 @@
 #ifndef AUTHPLUGINIF_H
 #define AUTHPLUGINIF_H

+#include <QtCore/qglobal.h>
 #include <QtCore/qobject.h>
 #include <QtCore/qpointer.h>
 #include <QtCore/qplugin.h>
@@ -37,9 +38,10 @@
 #include <SignOn/uisessiondata.h>
 #include <SignOn/signonerror.h>

+#include <QStringList>
+
 QT_BEGIN_NAMESPACE
 class QString;
-class QStringList;
 class QByteArray;
 class QVariant;
 QT_END_NAMESPACE
